lib/matplotlib/mathtext.py

```diff
@@ -660,6 +660,7 @@ class BakomaFonts(TruetypeFonts):
 
     def __init__(self, *args, **kwargs):
         self._stix_fallback = StixFonts(*args, **kwargs)
+        self.name = "Computer Modern"
```

Comment:
Maybe I'm doing it wrong, but the only reasonable way I found was `self.cm_fallback.default_font_prop.get_family()` but that just returns `['sans-serif']` which doesn't help here.

Reply:
Or fontmap which returns sth like ```{'cal': '/home/anovak/matplotlib/lib/matplotlib/mpl-data/fonts/ttf/cmsy10.ttf',
...
'cmsy10': '/home/anovak/matplotlib/lib/matplotlib/mpl-
data/fonts/ttf/cmex10.ttf', 'cmex10': '/home/anovak/matplotlib/lib/matplotlib/mpl-data/fonts/ttf/cmex10.ttf'}``` which would need some extra matching overhead to spit out a useful message.

Reply:
I was basically considering something like
```python
                if (fontname in ('it', 'regular')
                        and isinstance(self.cm_fallback, StixFonts)):
                    fontname = 'rm'

                g = self.cm_fallback._get_glyph(
                    fontname, font_class, sym, fontsize)
                _log.warning("Substituting with a symbol from {}".format(g[0].family_name))
                # or .postscript_name
                return g
```
As it turns out, while this would work for stix fallback and most other fonts (I guess), this would fail specifically for computer modern fallback because cmmi10.ttf has poor metadata (it lists its family as "cmmi10"...), but I guess you can always special-case cm; at least this should work for "all" other fallbacks...

Reply:
Ah nice, that's quite fun, because you can report per symbol. Checkout:
```
import matplotlib as mpl
test_str = r'a$b\AA\Finv\breve\gimel\succnapprox$'

mpl.rcParams['mathtext.fontset'] = 'custom'
mpl.rcParams['mathtext.rm'] = 'Courier New'
mpl.rcParams['mathtext.it'] = 'Courier New:italic'
mpl.rcParams['mathtext.bf'] = 'Courier New:bold'

fig, ax = plt.subplots(figsize=(8, 8))

mpl.rcParams['mathtext.fallback'] = 'cm'
txt = fig.text(.5, .5, test_str, fontsize=40, ha='center')
```
Output:
```
Substituting with a \Finv from STIXGeneral
Substituting with a \gimel from STIXGeneral
Substituting with a \combiningbreve from Computer Modern
Substituting with a \succnapprox from STIXGeneral
Substituting with a \Finv from STIXGeneral
Substituting with a \gimel from STIXGeneral
Substituting with a \combiningbreve from Computer Modern
Substituting with a \succnapprox from STIXGeneral
```
Can't figure out why is it doing too loops though.

Reply:
Probably once to compute the text metrics (for alignment) and once for the actual rendering.
You can avoid the duplication by using warnings.warn (or rather cbook._warn_external), which I guess seems ok here.

Reply:
@anntzer yeah, warnings fail tests

Reply:
You can use recwarn to catch them.

Reply:
My point was that you actually want to emit a test here, but *in the tests* you can check that the warning is correctly thrown using pytest.warns.

Reply:
better just go back to logging, than changing all those tests

Reply:
that's fine with me too, I don't think the extra logging is too bad(?)